src/lib/comms/include/sol-netctl.h

```diff
@@ -0,0 +1,476 @@
+};
+
+/**
+ * @brief Service monitor callback used to inform a service changed
+ *
+ * @param data the user data
+ * @param service the monitor service
+ */
+typedef void (*sol_netctl_service_monitor_cb)
+    (void *data,
+    const struct sol_netctl_service *service);
+
+/**
+ * @brief Manager monitor callback used to inform a manager updated
+ *
+ * @param data the user data
+ */
+typedef void (*sol_netctl_manager_monitor_cb)(void *data);
+
+/**
+ * @brief error monitor callback used to inform a call result
+ *
+ * @param data the user data
+ * @param service the activity service
+ * @param error the call result
+ */
+typedef void (*sol_netctl_error_monitor_cb)
+    (void *data, const struct sol_netctl_service *service,
+    unsigned int error);
+
+/**
+ * @brief Gets the service name
+ *
+ * This function gets the name for the netctl service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the name is desired
+ *
+ * @return The name of the services on success, NULL on failure.
+ */
+const char *sol_netctl_service_get_name(
+    const struct sol_netctl_service *service);
+
+/**
+ * @brief Gets the service state
+ *
+ * This function gets the state for the netctl service,
+ * after the service is to give by service monitor.
+ *
+ * @note SOL_NETCTL_SERVICE_STATE_REMOVE is used to show
+ * the service has been removed. When it is notified,
+ * the service has been removed in the system.
+ *
+ * @param service The service structure which the state is desired
+ *
+ * @return The state of the services on success,
+ * SOL_NETCTL_SERVICE_STATE_UNKNOWN on failure.
+ */
+enum sol_netctl_service_state sol_netctl_service_get_state(
+    const struct sol_netctl_service *service);
+
+/**
+ * @brief Gets the service type
+ *
+ * This function gets the type for the netctl service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the type is desired
+ *
+ * @return The type of the services on success, NULL on failure.
+ */
+const char *sol_netctl_service_get_type(
+    const struct sol_netctl_service *service);
+
+/**
+ * @brief Gets the service network address
+ *
+ * This function gets the network address for the netctl service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the network address is desired
+ * @param link The retrieved content
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_netctl_service_get_network_address(
+    const struct sol_netctl_service *service, struct sol_network_link **link);
+
+/**
+ * @brief Gets the service strength
+ *
+ * This function gets the service strength for the netctl service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the service strength is
+ * desired
+ *
+ * @return 0-100 on success, -errno on failure.
+ */
+int32_t sol_netctl_service_get_strength(
```

Comment:
I've asked to keep the int32_t, in the beginning I thought it could go over 100, then fixing the 32 bits in size was the way to go. Now it's clear it's not needed. But doesn't hurt to keep that.